### PR TITLE
dm_sa OK

### PR DESCRIPTION
**Pull request by kyleburnette**

### CODE REVIEW

File: src/overlays/actors/ovl_Dm_Sa/z_dm_sa.h [hunk 1]
```diff
@@ -5,9 +5,19 @@
 
 struct DmSa;
 
+typedef void (*DmSaActionFunc)(struct DmSa*, GlobalContext*);
+
 typedef struct DmSa {
     /* 0x000 */ Actor actor;
-    /* 0x144 */ char unk_144[0x1B0];
+    /* 0x144 */ SkelAnime skelAnime;
+    /* 0x188 */ char unk194[0x108];
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Dm_Sa/z_dm_sa.h:13]: I generally like to keep these unk fields UNK_TYPE1 if it still matches, to signal to the reader of this code that we don't know the signedness of this data.

**Review by Kenix3 — CHANGES_REQUESTED** [review]
Looks good to me, one small nitpick.

File: ctx.c [hunk 2]
```diff
@@ -0,0 +1,10992 @@
+typedef signed char s8;
```
Comment (Rozelette) [inline ctx.c:1]: This file shouldn't be included

File: src/overlays/actors/ovl_Dm_Sa/z_dm_sa.c [hunk 3]
```diff
@@ -19,28 +20,102 @@ const ActorInit Dm_Sa_InitVars = {
...
+extern AnimationHeader D_0600CC94;
+
+static ActorAnimationEntry D_80A2ED00[] = { { &D_0600CC94, 1.0f, 0, -1.0f, 0, 0 } };
+
+void func_80A2E960(SkelAnime* arg0, ActorAnimationEntry* animations, u16 index) {
+    f32 frameCount;
+    animations += index;
+
+    if (animations->frameCount < 0.0f) {
+        frameCount = (f32)SkelAnime_GetFrameCount(animations->animation);
```
Comment (Rozelette) [inline src/overlays/actors/ovl_Dm_Sa/z_dm_sa.c:36]: Is this cast needed?

File: src/overlays/actors/ovl_Dm_Sa/z_dm_sa.c [hunk 4]
```diff
@@ -19,28 +20,102 @@ const ActorInit Dm_Sa_InitVars = {
...
+}
+
+void func_80A2EB44(s32 arg0, s32 arg1, s32 arg2) {
+}
+
+Gfx* func_80A2EB58(GraphicsContext* gfxCtx, u32 alpha) {
+    Gfx* dList;
+    Gfx* dListHead;
+
+    dList = dListHead = GRAPH_ALLOC(gfxCtx, sizeof(Gfx) * 2);
```
Comment (Rozelette) [inline src/overlays/actors/ovl_Dm_Sa/z_dm_sa.c:85]: I saw the discussion about this bug, so I'd put
```suggestion
    dList = dListHead = GRAPH_ALLOC(gfxCtx, sizeof(Gfx) * 2); //! @bug this does not allocate enough for 3 Gfx commands
```

File: tools/gen_mips_to_c_context.py:Zone.Identifier [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[ZoneTransfer]
```
Comment (Rozelette) [inline tools/gen_mips_to_c_context.py:Zone.Identifier:1]: Remove this file

**Review by Rozelette — CHANGES_REQUESTED** [review]
LGTM except for nits and the extraneous files

**Review by Rozelette — APPROVED** [review]
LGTM

**Review by Kenix3 — APPROVED** [review]
Looks good to me.

**Pull request merged**